language/puppet-functions.md

```diff
@@ -51,29 +51,60 @@ possible to call a puppet function with a lambda, and possibly also type a param
 
 Function Definition - Scope & Autoloading
 ---
-Puppet Functions are autoloaded from modules. They are located under <module-root>/functions.
-The filename must match the simple (non namespaced name part) of the function. Thus, a function 'min' in module 'testmodule' is placed like this:
+
+#### Autoloading
+
+Puppet Functions are autoloaded from modules and the environment when located under <module-root>/functions or <environment-root>/functions respectively.
```

Comment:
> Namespaced Puppet functions are autoloaded...

Better to specify like that right up front, since that is the only subset of functions affected by autoloading.